File: Week 01/id_437/283.移动零.js
/*
 * @lc app=leetcode.cn id=283 lang=javascript
 *
 * [283] 移动零
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
  let j = 0
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== 0) {
      nums[j++] = nums[i]
    }
  }
  for (let i = j; i < nums.length; i++) {
    nums[i] = 0
  }
  return nums
};
// @lc code=end

File: Week 01/id_477/283.移动零.go
package leetcode

/*
 * @lc app=leetcode.cn id=283 lang=golang
 *
 * [283] 移动零
 */

// @lc code=start
func moveZeroes(nums []int) {
	// j := 0
	// for i := 0; i < len(nums); i++ {
	// 	if nums[i] != 0 {
	// 		nums[j] = nums[i]
	// 		j++
	// 	}
	// }
	// for j < len(nums) {
	// 	nums[j] = 0
	// 	j++
	// }

	j := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[j], nums[i] = nums[i], nums[j]
			j++
		}
	}
}

// @lc code=end
